store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaOverlapTest.java

```diff
@@ -154,14 +160,19 @@ public void shouldOverwriteVisibilityPropertyWhenConflictingSchemasMerge() throw
         assertThat(schema.validate().isValid())
                 .withFailMessage(schema.validate().getErrorString())
                 .isTrue();
-        assertThat(schema.getVisibilityProperty()).isEqualTo(PROPERTY_1);
-        assertThat(schema.toString())
-                .isNotEqualTo(GRAPH_A_SCHEMA.toString())
-                .isEqualTo(graphBSchema.toString());
+        assertThat(schema.getVisibilityProperty())
+                .withFailMessage("After a recovery from a collision with Visibility Property, the property should be set to Null")
+                .isNull();
+        assertThat(schema)
+                .hasToString(new Schema.Builder(GRAPH_A_SCHEMA)
```

Comment:
Rather than checking the `toStrings` would it make more sense to assert they are the same?

Reply:
yes, but I think when this errors it shows you the strings, which is nice. I think that's why it's there.